test/e2e/app_suffix_test.go

```diff
@@ -100,21 +100,20 @@ func TestAppSuffix_AppExists_MigrationEnabled(t *testing.T) {
 		NewMissingClusterResource(t, "configmap", newName, env.Namespace, kubectl)
 	})
 
-	logger.Section("name already contains app suffix", func() {
-		existingName := name + app.AppSuffix
+	logger.Section("name contains app suffix for migrated app", func() {
```

Comment:
This test was previously capturing what I saw as desired behaviour:

1. `KAPP_FQ_CONFIGMAP_NAMES=False kapp deploy -a foo.apps.k14s.io -f ...` ==> configmap = foo.apps.k14s.io
2. `KAPP_FQ_CONFIGMAP_NAMES=True kapp deploy -a foo.apps.k14s.io -f ...` ==> configmap = foo.apps.k14s.io.apps.k14s.io

Looks like this test now checks:
1. `KAPP_FQ_CONFIGMAP_NAMES=True kapp deploy -a foo -f ...` ==> configmap = foo.apps.k14s.io
2. `KAPP_FQ_CONFIGMAP_NAMES=True kapp deploy -a foo.apps.k14s.io -f ...` ==> configmap = foo.apps.k14s.io

---

So we are somewhat exposing the internal configmap suffix name to the user now, if you use this suffix - we are making an assumption you have migrated your app. Feels odd, that was the purpose of the annotation - to remove this value judgement, and assume the user given app_name is what they want their app to be named.

Reply:
Yeah, it definitely is a bit weird. I will switch to the other approach.